cpg-language-python/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/python/PythonValueEvaluator.kt

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (c) 2025, Fraunhofer AISEC. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *                    $$$$$$\  $$$$$$$\   $$$$$$\
+ *                   $$  __$$\ $$  __$$\ $$  __$$\
+ *                   $$ /  \__|$$ |  $$ |$$ /  \__|
+ *                   $$ |      $$$$$$$  |$$ |$$$$\
+ *                   $$ |      $$  ____/ $$ |\_$$ |
+ *                   $$ |  $$\ $$ |      $$ |  $$ |
+ *                   \$$$$$   |$$ |      \$$$$$   |
+ *                    \______/ \__|       \______/
+ *
+ */
+package de.fraunhofer.aisec.cpg.frontends.python
+
+import de.fraunhofer.aisec.cpg.analysis.ValueEvaluator
+import de.fraunhofer.aisec.cpg.graph.HasOperatorCode
+import de.fraunhofer.aisec.cpg.graph.Node
+import de.fraunhofer.aisec.cpg.graph.statements.expressions.Expression
+import de.fraunhofer.aisec.cpg.graph.statements.expressions.InitializerListExpression
+import de.fraunhofer.aisec.cpg.graph.statements.expressions.Reference
+import de.fraunhofer.aisec.cpg.passes.reconstructedImportName
+import kotlin.math.pow
+
+/** An extended version of the [ValueEvaluator] that supports Python-specific operations. */
+class PythonValueEvaluator : ValueEvaluator() {
+    override val cannotEvaluate: (Node?, ValueEvaluator) -> Any?
+        get() = { node, evaluator ->
+            if (node is InitializerListExpression) {
+                // We can evaluate initializer lists if all elements are constant
+                val values = node.initializers.map { evaluator.evaluate(it) }
+                if (values.all { it is Number }) {
+                    values
+                } else {
+                    super.cannotEvaluate(node, evaluator)
+                }
+            } else {
+                super.cannotEvaluate(node, evaluator)
+            }
+        }
+
+    override fun handleReference(expr: Reference, depth: Int): Any? {
+        // We need to handle sys.platform and sys.version_info specially, since it is often used in
+        // a pre-processor macro-style, and we want to replace this with the actual value (if we
+        // have it). This allows us to dynamically prune if-branches based on constant evaluation.
+        return when {
+            expr.reconstructedImportName.toString() == "sys.platform" ->
+                PythonLanguageFrontend.SysInfo.platform ?: super.handleReference(expr, depth)
```

Comment:
Could we place this information as a node (maybe via `Concept`?) somewhere in the `TranslationResult` or probably better in the `Component`? Even if we're in the python module, I somehow would prefer accessing this information through the graph.

This also affects the next lines of course.

Reply:
Hmm good point. I will think about a solution.

Reply:
I added the `SystemInformation` as an overlay node to the current TU(s)